keps/sig-scheduling/1819-scheduler-extender/README.md

```diff
@@ -0,0 +1,299 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes scheduler schedules based on resources managed by Kubernetes.
+Scheduling based on opaque resource counting helps extending this further. But
+when there is a need for contextual scheduling for resources managed outside of
+kubernetes, there is no mechanism to do it today.
+
+The proposal is to make kubernetes scheduler extensible by adding the capability
+to make http calls out to another endpoint to help achieve this functionality.
+
+## Motivation
+
+There are three ways to add new scheduling rules to Kubernetes:
+
+- Update existing or add new scheduler plugins and recompiling
+- Implementing your own scheduler process that runs instead of, or alongside of, the standard Kubernetes scheduler
+- Implementing a "scheduler extender" process that the standard Kubernetes scheduler calls out to as a final pass when making scheduling decisions
+
+This approach is needed for use cases where scheduling decisions need to be
+made on resources not directly managed by the standard Kubernetes scheduler.
+The extender helps make scheduling decisions based on such resources.
+
+### Goals
+
+- Able to extend an existing scheduler by registering an external webhook
+
+### Non-Goals
+
+- Solve limitations due to running extender as a separate process, e.g. cache
+  inconsistency, performance
+
+## Design Details
+
+When scheduling a pod, the extender allows an external process to filter and
+prioritize nodes. Two separate http/https calls are issued to the extender, one
+for "filter" and one for "prioritize" actions. If the pod cannot be scheduled,
+the scheduler will try to preempt lower priority pods from nodes and send them
+to extender "preempt" verb if configured. The extender can return a subset of
+nodes and new victims back to the scheduler. Additionally, the extender can
+choose to bind the pod to apiserver by implementing the "bind" action.
+
+### Configuration
+
+To use the extender, you must create a scheduler policy configuration file. The
```

Comment:
the wording and API policy will be deprecated, so let's just say "scheduler configuration file":

```suggestion
To use the extender, you must create a scheduler configuration file. The
```